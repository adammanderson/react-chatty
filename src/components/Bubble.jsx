import React from 'react'
import PropTypes from 'prop-types'
import marked from 'marked'
import styled from 'react-emotion'

const BubbleWrapper = styled.div`
  display: flex;
  flex-flow: column nowrap;
  margin-bottom: 1.3em;
  align-items: ${props => props.theme.bubble.find(b => b.type === props.type).align}
`

const BubbleContent = styled.span`
  padding: 0.9em 1.2em;
  max-width: 70%;
  line-height: 1.5em;
  border-radius: ${props => props.type === 'user' ? '1.5em 1.5em 0 1.5em;' : '1.5em 1.5em 1.5em 0;'}
  background: ${props => props.theme.bubble.find(b => b.type === props.type).background};
  color: ${props => props.theme.bubble.find(b => b.type === props.type).color};
  p {
    margin: 0;
  }
`

const Bubble = (props) => (
  <BubbleWrapper type={props.type}>
    <BubbleContent
      type={props.type}
      dangerouslySetInnerHTML={{__html: marked(props.content)}}
    />
  </BubbleWrapper>
)

Bubble.propTypes = {
  content: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['user', 'agent', 'waiting']).isRequired
}

export default Bubble
