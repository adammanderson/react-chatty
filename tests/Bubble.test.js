import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming'
import THEMES from '../src/themes'
import Bubble from '../src/components/Bubble';

test('Bubble renders with theme', () => {
  const component = renderer.create(
    <ThemeProvider theme={THEMES.light}>
      <Bubble
        content="Test"
        type="user"
      />
    </ThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
