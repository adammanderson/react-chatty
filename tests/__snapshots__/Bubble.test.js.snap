// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bubble renders with theme 1`] = `
<div
  className="css-ra4cbz e193rrgr0"
  type="user"
>
  <span
    className="css-5i98md e193rrgr1"
    type="user"
  >
    Test
  </span>
</div>
`;
